sdk/data/azcosmos/cosmos_patch_operations.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// PatchOperationType defines supported values for operation types in Patch Document.
+type patchOperationType string
+
+const (
+	// Represents a patch operation Add.
+	patchOperationTypeAdd patchOperationType = "add"
+	// Represents a patch operation Replace.
+	patchOperationTypeReplace patchOperationType = "replace"
+	// Represents a patch operation Remove.
+	patchOperationTypeRemove patchOperationType = "remove"
+	// Represents a patch operation Set.
+	patchOperationTypeSet patchOperationType = "set"
+	// Represents a patch operation Increment.
+	patchOperationTypeIncrement patchOperationType = "incr"
+)
+
+// PatchOperation represents individual patch operation.
+type patchOperation struct {
+	Op    patchOperationType `json:"op"`
+	Path  string             `json:"path"`
+	Value interface{}        `json:"value,omitempty"`
+}
+
+// PatchOperations represents the patch request.
+// See https://learn.microsoft.com/azure/cosmos-db/partial-document-update
+type PatchOperations struct {
+	condition  *string
+	operations []patchOperation
+}
+
+// MarshalJSON implements the json.Marshaler interface
+func (o PatchOperations) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString("{")
+	if o.condition != nil {
+		buffer.WriteString(fmt.Sprintf("\"condition\":\"%s\",", *o.condition))
+	}
+	buffer.WriteString("\"operations\":[")
+	for i, operation := range o.operations {
+		if i > 0 {
+			buffer.WriteString(",")
+		}
+		operationBytes, err := json.Marshal(operation)
+		if err != nil {
+			return nil, err
+		}
+		buffer.Write(operationBytes)
+	}
+	buffer.WriteString("]}")
+	return buffer.Bytes(), nil
+}
+
+// SetCondition sets condition for the patch request.
+func (p *PatchOperations) SetCondition(condition string) {
+	p.condition = &condition
+}
+
+// AppendReplace appends a replace operation to the patch request.
+func (p *PatchOperations) AppendReplace(path string, value interface{}) {
```

Comment:
Nit: Go 1.18 introduced keyword `any` which is an alias for `interface{}` which we've been trying to use as it's shorter/cleaner.

Reply:
Perfect. Done :)